### PR TITLE
Add Primary Person Variant to Relationships

### PR DESCRIPTION
**Pull request by iwootten**
### What is the context of this PR?

Currently during relationship definition, we have no way of displaying question variants for a primary person. 

### How to review 
This adds a new test_relationship_primary schema which includes a question variant for the primary person. To support display of the variant a new "when" comparison needs to be added to allow comparisons against location.

This needs to be run with: https://github.com/ONSdigital/eq-schema-validator/pull/143

### CODE REVIEW

File: scripts/test_schemas.sh
Comment (iwootten) [inline scripts/test_schemas.sh]: Update the branch once validator PR is merged

**Comment by iwootten** [comment]
[Add primary person to relationships (M)](https://trello.com/c/B20tHhR5/3067-add-primary-person-to-relationships-m)

File: app/questionnaire/rules.py
Comment (JamesGardiner) [inline app/questionnaire/rules.py]:
```Python
try:
    value = getattr(list_store.get(when_rule['list']), when_rule['id_selector'])
except KeyError:
    value = get_list_count(list_store, when_rule['list'])
```

**Review by JamesGardiner — CHANGES_REQUESTED** [review]
LGTM - one minor change request

File: app/questionnaire/rules.py
Comment (iwootten) [reply app/questionnaire/rules.py]: Neither one of these is an exception, where this change implies one of them is. I think the original is more readable.

Comment (JamesGardiner) [reply app/questionnaire/rules.py]: I disagree - the suggested change is an idiomatic approach to handling mapping access control flow.

Comment (iwootten) [reply app/questionnaire/rules.py]: I'm sorry but I still maintain that my change is more readable and consistent with what is already there. The project hasn't ever prescribed the EAFP approach as far as I'm aware, so I'm not sure why this should block the PR?

Comment (bitdivision) [reply app/questionnaire/rules.py]: I don't really see much of a difference in readability, but the if pattern does keep this consistent with the rest of runner. 

We should probably avoid having this argument on all PRs until we decide on an approach with the whole team.

**Review by bitdivision — APPROVED** [review]
Works well, and the changes look sensible.

**Review by JamesGardiner — APPROVED** [review]

File: app/questionnaire/rules.py
Comment (bitdivision) [inline app/questionnaire/rules.py:217]: :+1:

**Pull request merged**
